lib/msal-common/src/account/ClientCredentials.ts

```diff
@@ -3,11 +3,14 @@
  * Licensed under the MIT License.
  */
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type ClientAssertionCallbackFunction = (...args: any[]) => string;
+
 /**
  * Client Assertion credential for Confidential Clients
  */
 export type ClientAssertion = {
-    assertion: string;
```

Comment:
breaking change.

Reply:
ah. The GitHub issue says "instead of".

Reply:
sorry, yeah supporting both can be an option too :)

Reply:
yeah so what about a constructor overload?

Reply:
[I don't think TypeScript support that](https://github.com/microsoft/TypeScript/issues/54397). Do you have evidence otherwise?

Reply:
Anyways, I'd like to Bogdan to leave his feedback on the implementation I just finished up today before I look into that.

Reply:
Not a JS / TS expert, I think this is ok. We welcome your feedback @KarishmaGhiya